common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -331,30 +398,41 @@ def validate(self):
             )
             return validation
         lib_tools = self.runtime.service(self, 'library_tools')
+        matching_children_count = 0
         for library_key, version in self.source_libraries:
-            latest_version = lib_tools.get_library_version(library_key)
-            if latest_version is not None:
-                if version is None or version != latest_version:
-                    validation.set_summary(
-                        StudioValidationMessage(
-                            StudioValidationMessage.WARNING,
-                            _(u'This component is out of date. The library has new content.'),
-                            action_class='library-update-btn',  # TODO: change this to action_runtime_event='...' once the unit page supports that feature.
-                            action_label=_(u"↻ Update now")
-                        )
-                    )
-                    break
-            else:
-                validation.set_summary(
-                    StudioValidationMessage(
-                        StudioValidationMessage.ERROR,
-                        _(u'Library is invalid, corrupt, or has been deleted.'),
-                        action_class='edit-button',
-                        action_label=_(u"Edit Library List")
-                    )
-                )
+            if not self._validate_library_version(validation, lib_tools, version, library_key):
                 break
 
+            library = lib_tools.get_library(library_key)
+            children_matching_filter = lib_tools.get_filtered_children(library, self.capa_type)
+            # get_filtered_children returns generator, so can't use len.
+            # And we don't actually need those children, so no point of constructing a list
+            matching_children_count += sum(1 for child in children_matching_filter)
+
+        if matching_children_count == 0:
+            self._set_validation_error_if_empty(
+                validation,
+                StudioValidationMessage(
+                    StudioValidationMessage.WARNING,
+                    _(u'There are no matching problem types in the specified libraries.'),
+                    action_class='edit-button',
+                    action_label=_(u"Select another problem type")
+                )
+            )
+
+        if matching_children_count < self.max_count:
+            self._set_validation_error_if_empty(
+                validation,
+                StudioValidationMessage(
+                    StudioValidationMessage.WARNING,
+                    _(u'The specified libraries are configured to fetch {count} problems, '
+                      u'but there are only {actual} matching problems.')
+                    .format(actual=matching_children_count, count=self.max_count),
+                    action_class='edit-button',
+                    action_label=_(u"Edit configuration")
```

Comment:
Sorry, seeing this in the UI made me realize the action label should have sentence punctuation. Please add period at end: "Edit the library configuration."